text/0000-deletable-immutable-data/0000-deletable-immutable-data.md

```diff
@@ -0,0 +1,68 @@
+# ImmutableData deletion support
+- Status: proposed
+- Type: feature
+- Related components: `safe_core`, `safe_vault`, `safe_launcher`
+- Start Date: 04-August-2016
+
+## Summary
+This is an attempt to present the benefits of being able to delete `ImmutableData` and show a possible approach to do it.
+
+## Conventions
+- The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
+
+## Motivation
+`ImmutableData` is considered to be a network-owned data. As such users of SAFE Network can create it but never delete it. Moreover by their very nature `ImmutableData` are immutable and this immutablility is baked into the fact that the location where `ImmutableData` are stored (also called the `name` of `ImmutableData`) is derived from the hash of its content (also referred to as the `body`). Any change in the content will consequently produce a new name and the network will store it in a different place. Also `ImmutableData` is de-duplicated, i.e. if several users upload the same `ImmutableData`, they will all be charged but the network itself will have only one primary copy of it. However there are several cases where de-duplication is very unlikely, for instance if the content was encrypted or if the content was a representation of versioned directory listing of a particular user etc. In these cases when the user updates the content, they might no longer have any need for the previous content. There are also cases where one would like to delete a previously created `ImmutableData`. Currently there is neither a mechanism nor an incentive to delete unneeded `ImmutableData`. In practice we find such accumulation of never required data highly wasteful. Each churn event must account for exchange of all such data too as the network has no knowledge of what is required and what is not. Being able to delete `ImmutableData` would be nice to have and should be incentivised.
```

Comment:
Wouldn't _encrypted_ signify privacy ?
`privately encrypted` seems to suggest that it could be `publicly encrypted` too which would be a vague statement.